usd/src/usd_parser/USDLinks.cc

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ *
+*/
+
+#include "USDLinks.hh"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#pragma push_macro ("__DEPRECATED")
+#undef __DEPRECATED
+#include "pxr/usd/usdPhysics/rigidBodyAPI.h"
+#pragma pop_macro ("__DEPRECATED")
+
+#include <ignition/common/Filesystem.hh>
+
+#include <ignition/math/Inertial.hh>
+
+#include "sdf/Link.hh"
+
+#include "sdf/usd/usd_parser/USDTransforms.hh"
+
+namespace sdf
+{
+// Inline bracket to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+void GetInertial(
+  const pxr::UsdPrim &_prim,
+  ignition::math::Inertiald &_inertial)
+{
+  float mass;
+  pxr::GfVec3f centerOfMass;
+  pxr::GfVec3f diagonalInertia;
+  pxr::GfQuatf principalAxes;
+
+  ignition::math::MassMatrix3d massMatrix;
+
+  if (_prim.HasAPI<pxr::UsdPhysicsRigidBodyAPI>())
+  {
+    if (pxr::UsdAttribute massAttribute =
+      _prim.GetAttribute(pxr::TfToken("physics:mass")))
+    {
+      massAttribute.Get(&mass);
+
+      if (pxr::UsdAttribute centerOfMassAttribute =
+        _prim.GetAttribute(pxr::TfToken("physics:centerOfMass")))
+      {
+        centerOfMassAttribute.Get(&centerOfMass);
+      }
+
+      if (pxr::UsdAttribute diagonalInertiaAttribute =
+        _prim.GetAttribute(pxr::TfToken("physics:diagonalInertia")))
+      {
+        diagonalInertiaAttribute.Get(&diagonalInertia);
+      }
+
+      if (pxr::UsdAttribute principalAxesAttribute =
+        _prim.GetAttribute(pxr::TfToken("physics:principalAxes"))) {
+        principalAxesAttribute.Get(&principalAxes);
+      }
+
+      // Added a diagonal inertia to avoid crash with the physics engine
+      if (diagonalInertia == pxr::GfVec3f(0, 0, 0))
+      {
+        diagonalInertia = pxr::GfVec3f(0.0001, 0.0001, 0.0001);
+      }
+
+      // Added a massto avoid crash with the physics engine
+      if (mass < 0.0001)
+      {
+        mass = 0.1;
+      }
+
+      massMatrix.SetMass(mass);
+      massMatrix.SetDiagonalMoments(
+        ignition::math::Vector3d(
+          diagonalInertia[0],
+          diagonalInertia[1],
+          diagonalInertia[2]));
+
+      _inertial.SetPose(ignition::math::Pose3d(
+          ignition::math::Vector3d(
+            centerOfMass[0], centerOfMass[1], centerOfMass[2]),
+          ignition::math::Quaterniond(1.0, 0, 0, 0)));
+
+      _inertial.SetMassMatrix(massMatrix);
+    }
+  }
+  else
+  {
+    auto parent = _prim.GetParent();
+    if (parent)
+    {
+      GetInertial(parent, _inertial);
+    }
+  }
+}
+
+void ParseUSDLinks(
+  const pxr::UsdPrim &_prim,
+  const std::string &_nameLink,
+  sdf::Link *_link,
+  USDData &_usdData)
+{
+  std::string primNameStr = _prim.GetPath().GetName();
+  std::string primPathStr = pxr::TfStringify(_prim.GetPath());
+  std::string primType = _prim.GetPrimTypeInfo().GetTypeName().GetText();
+
+  std::pair<std::string, std::shared_ptr<USDStage>> data =
+    _usdData.FindStage(primNameStr);
```

Comment:
I don't see the `data` variable being used anywhere else in this method. Can this call to `FindStage` be removed? I'm not sure that it's needed.

Reply:
removed